pkg/executor/push.go

```diff
@@ -99,11 +104,61 @@ var (
 	checkRemotePushPermission = remote.CheckPushPermission
 )
 
+// CreateECRRepo creates the ECR repository if it doesn't exist
+func CreateECRRepo(opts *config.KanikoOptions) error {
```

Comment:
Most of the logic is duplicated from the `CheckPushPermissions` function, may be the common logic can be refactored.